ext/mysqli/tests/mysqli_report.phpt

```diff
@@ -267,7 +267,12 @@ require_once('skipifconnectfailure.inc');
 			!mysqli_query($link, 'DELETE FROM test WHERE id > 50', MYSQLI_USE_RESULT))
 		printf("[033] [%d] %s\n", mysqli_errno($link), mysqli_error($link));
 
-	$tmp = mysqli_thread_safe($link);
+	try {
+	    mysqli_thread_safe($link);
+	} catch (ArgumentCountError $exception) {
+	    print($exception->getMessage() . "\n");
+	}
```

Comment:
You can just drop this part.

Reply:
If we're going to disallow passing a `mysqli` instance to `mysqli_thread_safe()`, we should also mark the associtated `mysqli::thread_safe()` as static. Same for `mysqli::get_client_info()`. See also https://bugs.php.net/78790.

Reply:
@cmb69 Thank you for your help on the issue. Does it need any discussion in order to declare them static or can I go ahead and create a PR?

Reply:
I think this should be discussed, at least. After all, even the [documentation states](https://www.php.net/mysqli_get_client_info) that `mysqli_get_client_info()` is supposed to be called with a `mysqli` argument, so this might be quite some BC break. :(

Reply:
If it's documented, let's go with what the docs say it should be.

Reply:
Problem: if we enforce the mysqli parameter, likely some code which doesn't pass would break. Also docs are inconsistent regarding `mysqli_get_client_info()` and `mysqli_thread_safe()` (the latter is parameterless). This is rather messy :(